### PR TITLE
RIA-6896 New import dialog

### PR DESCRIPTION
**Pull request by AlexanderChernyaev**
https://smartbear.atlassian.net/browse/RIA-6896
https://github.com/SmartBear/readyapi-free-os/pull/639

### CODE REVIEW

File: src/main/java/com/smartbear/plugins/swaggerhub/PluginConfig.java [hunk 1]
```diff
@@ -5,12 +5,9 @@
 import com.eviware.soapui.plugins.PluginDependencies;
 import com.eviware.soapui.plugins.PluginDependency;
 
-@PluginConfiguration(groupId = "com.smartbear.plugins", name = "SwaggerHub ReadyAPI Plugin", version = "1.2.0",
+@PluginConfiguration(groupId = "com.smartbear.plugins", name = "SwaggerHub ReadyAPI Plugin", version = "1.3.0",
```
Comment (antone-sb) [inline src/main/java/com/smartbear/plugins/swaggerhub/PluginConfig.java]: Could you set minimumReadyApiVersion too as I believe it won't work with ReadyAPI 2.3 and less.

File: src/main/java/com/smartbear/plugins/swaggerhub/ImportFromHubDialog.java [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+
+
+    protected void buildDialog() {
+        initModality(Modality.APPLICATION_MODAL);
+        setDialogPane(UISupport.getDialogPaneWithPlatformButtonOrder());
+        Scene scene = getDialogPane().getScene();
+        Stage stage = (Stage) scene.getWindow();
+        stage.getIcons().add(new Image(PRODUCT_ICON_PATH));
+        setResizable(false);
+        setTitle("Import From SwaggerHub");
```
Comment (antone-sb) [inline src/main/java/com/smartbear/plugins/swaggerhub/ImportFromHubDialog.java]: According to https://capitalizemytitle.com/, from shouldn't be capitalized.

File: src/main/java/com/smartbear/plugins/swaggerhub/ImportFromHubDialog.java [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+            if (StringUtils.isNotEmpty(login) && StringUtils.isNotEmpty(password) && searchInMyHub.isSelected()) {
+                uri = PluginConfig.SWAGGERHUB_API + "?filter=user";
+                try {
+                    getApiKey(password, login);
+                } catch (Exception e) {
+                    return;
+                }
+                importPrivate = true;
+            } else {
+                uri = PluginConfig.SWAGGERHUB_API + "?limit=50";
```
Comment (antone-sb) [inline src/main/java/com/smartbear/plugins/swaggerhub/ImportFromHubDialog.java]: Make 50 a constant field.

File: src/main/java/com/smartbear/plugins/swaggerhub/ImportFromHubDialog.java [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+        label.getStyleClass().add("default-text");
+        return label;
+    }
+
+    private void setTooltip(String text, Control component) {
+        Tooltip tooltip = new Tooltip(text);
+        component.setTooltip(tooltip);
+    }
+
+    private void getApiKey(String password, String login) throws Exception {
```
Comment (antone-sb) [inline src/main/java/com/smartbear/plugins/swaggerhub/ImportFromHubDialog.java]: I think login, password is more intuitive parameter order.

File: src/main/java/com/smartbear/plugins/swaggerhub/ImportFromHubDialog.java [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+
+            int selectedVersion = model.getVersionCombo().getSelectionModel().getSelectedIndex();
+
+            String version = descriptor.versions[selectedVersion];
+            if (version.startsWith("*") || version.startsWith("-")) {
+                version = version.substring(1).trim();
+            }
+
+            String url = swaggerUrl.substring(0, swaggerUrl.lastIndexOf('/')) + "/" + version;
+            System.out.println("Attempting to import Swagger from [" + url + "]");
```
Comment (antone-sb) [inline src/main/java/com/smartbear/plugins/swaggerhub/ImportFromHubDialog.java]: Change to logger or remove.

File: src/main/java/com/smartbear/plugins/swaggerhub/ImportFromHubDialog.java [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+            if (descriptor.isPrivate) {
+                CollectionUtils.addAll(result, importer.importSwagger(swaggerUrl, apiKey));
+            } else {
+                Collections.addAll(result, importer.importSwagger(swaggerUrl));
```
Comment (antone-sb) [inline src/main/java/com/smartbear/plugins/swaggerhub/ImportFromHubDialog.java]: Why do you use CollectionUtils in the first branch and Collections in the second?

File: src/main/java/com/smartbear/plugins/swaggerhub/ImportFromHubDialog.java [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+
+            if (StringUtils.isNotEmpty(extractedApiKey)) {
+                apiKey = extractedApiKey;
+            } else {
+                throw new Exception();
+            }
+
+        } catch (Exception e) {
+            Logging.logError(e, "Cannot retrieve API Key. Please check your credentials");
+            throw e;
```
Comment (antone-sb) [inline src/main/java/com/smartbear/plugins/swaggerhub/ImportFromHubDialog.java]: As you only log the exception, I think you can move logging to the place where you call this method and not catch the exception here.

File: src/main/java/com/smartbear/plugins/swaggerhub/ImportFromHubDialog.java [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+            httpPost.setEntity(params);
+            HttpResponse response = HttpClientSupport.getHttpClient().execute(httpPost);
+            String jsonResponse = new String(ByteStreams.toByteArray(response.getEntity().getContent()));
+            JSONObject jsonObject = new JSONObject(jsonResponse);
+            String extractedApiKey = jsonObject.getString("token");
+
+            if (StringUtils.isNotEmpty(extractedApiKey)) {
+                apiKey = extractedApiKey;
+            } else {
+                throw new Exception();
```
Comment (antone-sb) [inline src/main/java/com/smartbear/plugins/swaggerhub/ImportFromHubDialog.java]: It's a bad practice to throw an empty exception.

File: src/main/java/com/smartbear/plugins/swaggerhub/PublishToHubAction.java [hunk 9]
```diff
@@ -90,18 +88,8 @@ private boolean publishApi(RestService restService) throws IOException {
...
-            Method method;
-            try {
-                method = exporterClass.getMethod("exportToFolder", String.class, String.class, String.class, RestService[].class, String.class);
-            } catch (NoSuchMethodException e) {
-                method = exporterClass.getMethod("exportToFolder", String.class, String.class, String.class, AbstractRestService[].class, String.class);
-            }
-
-            String result = (String) method.invoke(exporter, Files.createTempDir().getAbsolutePath(),
-                    versionId, "json", new RestService[]{restService}, restService.getBasePath());
+            Swagger2Exporter exporter = new Swagger2Exporter(restService.getProject());
```
Comment (antone-sb) [inline src/main/java/com/smartbear/plugins/swaggerhub/PublishToHubAction.java:91]: Why is this Swagger2Exporter only and not OpenAPI3Exporter?

**Review by antone-sb — CHANGES_REQUESTED** [review]

File: src/main/java/com/smartbear/plugins/swaggerhub/PluginConfig.java (hunk 1 above)
Comment (AlexanderChernyaev) [reply src/main/java/com/smartbear/plugins/swaggerhub/PluginConfig.java]: Fixed

File: src/main/java/com/smartbear/plugins/swaggerhub/PublishToHubAction.java (hunk 9 above)
Comment (AlexanderChernyaev) [reply src/main/java/com/smartbear/plugins/swaggerhub/PublishToHubAction.java:91]: We have separate task RIA-7261

**Review by antone-sb — APPROVED** [review]

**Pull request merged**